ticket.sh

```
@@ -0,0 +1,2554 @@
+        current_args=$(tr '\0' ' ' < /proc/self/cmdline 2>/dev/null || echo "")
+    elif command -v ps >/dev/null 2>&1; then
+        # macOS/BSD: use ps command
+        current_args=$(ps -p $$ -o args= 2>/dev/null || echo "")
+    fi
+    
+    # Check if invoked via shell (bash, sh, zsh, etc.)
+    local shell_pattern='^(bash|sh|dash|zsh|fish|ksh|/.*/(bash|sh|dash|zsh|fish|ksh))[[:space:]]+'
+    
+    if [[ "$current_args" =~ $shell_pattern ]]; then
+        # Extract shell command
+        local shell_cmd=$(echo "$current_args" | sed -E 's/^([^[:space:]]+).*/\1/')
```

Shellcheck (0.10.0)

[warning] Declare and assign separately to avoid masking return values.

(SC2155)